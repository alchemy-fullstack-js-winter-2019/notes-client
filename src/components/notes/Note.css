.Note {
  display: block;
  margin: auto;

  h2 {
    padding: 0.2em 0;
    color: #B16FB1;
  }

  textarea {
    line-height: 1.5em;
  }
}
