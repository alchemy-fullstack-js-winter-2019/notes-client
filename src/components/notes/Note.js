import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FaPen } from 'react-icons/fa';
import styles from './Note.css';

function Note({ note }) {
  const { _id, title, body } = note;
  return (
    <article className={styles.Note}>
      <h2>{title}</h2>
      <pre>{body}</pre>
      <Link to={`/notes/${_id}/edit`}>
        <FaPen />
      </Link>
    </article>
  );
}

Note.propTypes = {
  note: PropTypes.object.isRequired
};

export default Note;
