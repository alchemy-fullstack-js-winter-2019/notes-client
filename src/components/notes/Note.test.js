import React from 'react';
import { shallow } from 'enzyme';
import Note from './Note';

describe('Note tests', () => {
  it('matches a snapshot', () => {
    const note = {
      _id: '1234',
      author: 'test.user',
      title: 'My Title',
      body: 'My Body'
    };
    expect(shallow(
      <Note note={note} />
    )).toMatchSnapshot();
  });


});
