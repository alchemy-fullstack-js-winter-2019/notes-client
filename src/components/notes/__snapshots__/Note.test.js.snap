// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note tests matches a snapshot 1`] = `
<article
  className="Note"
>
  <h2>
    My Title
  </h2>
  <pre>
    My Body
  </pre>
  <Link
    replace={false}
    to="/notes/1234/edit"
  >
    <FaPen />
  </Link>
</article>
`;
